add additional information about adding secrets (#70)
konflux-ci/docs: docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc

When you build your pipeline, you might want to add tasks that require **secrets** in order to access external resources.

> .Secrets fall into three catagories of when they can be added:

Reviewer: 1. Usually we don't mark such introductory sentences as titles (with . at the beginning), even if they serve as list titles. This sentence can remain a plain text.

2. Writers are discouraged from using the passive voice where it can be avoided. The preferred wording would be: ...of when you add them. Or even simpler: There're three categories of secrets depending on when you add them.

3. catagories > categories.

> . Before a component is added. If a secret is needed to access the source control platform (like xref:creating-secrets.adoc#creating-source-control-secrets[GitLab]), then the secret must be created before the component is created.

Reviewer: ```suggestion
. Before a component is added. If a secret is needed to access the source control platform (like xref:creating-secrets.adoc#creating-source-control-secrets[GitLab]), you must create a secret before you create the component.
```

The general rule of thumb for RH docs is to be as user-oriented as we can. Meaning we put 'you' everywhere to emphasize this is an action a user needs to do. But then it's not a requirement.

> . Before a build succeeds. Some artifact build tasks need specific secrets to be able to pull all of the content to include in the final artifact. These secrets like for xref:creating-secrets.adoc#creating-registry-pull-secrets[container registries] can be added after the component is created but must be provided before a successful build can occur.

Reviewer: ```suggestion
. Before a build succeeds. Some artifact build tasks need specific secrets to be able to pull all of the content to include in the final artifact. For example, you can add secrets for xref:creating-secrets.adoc#creating-registry-pull-secrets[container registries] after you create the component but they must be provided before a successful build can occur.
```
Same as on line 8, you can go for a more user-focused 'you'. And such texts are faster and easier to read than the ones with the passive voice all around.

> . After a component has been onboarded. These secrets are often used in tasks. The tasks included in the Konflux pipelines will not fail if a secret is not created properly. Instead, the task will just not run the code like with xref:creating-secrets.adoc#creating-task-input-secrets[snyk].

Reviewer: About xrefs.
Paths need to be given from the _pages_ family directory, e.g.
xref:how-tos/configuring/creating-secrets.adoc#creating-source-control-secrets[GitLab]

OR
Here we can also use in-page xrefs because we refer to the titles on the same page. This format should work (no spaces around a comma):

<<creating-source-control-secrets,GitLab>>
<<creating-registry-pull-secrets,container registries>>
<<creating-task-input-secrets,snyk>>

If Antora doesn't process them correctly for some reason, we can add ID's anywhere on the page like [#any-id-name] and then refer to it via <<any-id-name,text to show>>.

> . After a component has been onboarded. These secrets are often used in tasks. The tasks included in the Konflux pipelines will not fail if a secret is not created properly. Instead, the task will just not run the code like with xref:creating-secrets.adoc#creating-task-input-secrets[snyk].

Reviewer: Konflux > {ProductName}

== Creating task input secrets

> Some tasks may need secrets to be passed in order for them to be run properly. Consult the documentation for these tasks to understand the proper specification of the secrets required (for example the required keys/values to be present).

Reviewer: ```suggestion
Sometimes to run the tasks properly, you may need to pass secrets to these tasks. Consult the documentation for these tasks to understand the proper specification of the secrets required, for example the required keys/values.
```
Simplified grammar might add to the readability.

NOTE: One such task is the link:https://github.com/konflux-ci/build-definitions/tree/main/task/sast-snyk-check[sast-snyk-check] task that uses the third-party service link:https://snyk.io/[snyk] to perform static application security testing (SAST) as a part of the default {ProductName} pipeline. Use this procedure to upload your snyk.io token. Name the secret `sast_snyk_task` so that the snyk task in the {ProductName} pipeline will recognize it and use it.

.Procedure 

. In {ProductName}, from the left navigation menu, select **Secrets**.
. From the **Secrets** page, click **Add secret**.

> . For **secret type**, select **Key/value secret**

Reviewer: If that's a UI element / field name, we capitalize it: Secret type.

. Optional: Under **Labels**, add a label to tag or provide more context for your secret.
. Click **Add secret**.

== Creating registry pull secrets

> Some container builds may use parent images from registries that require authentication (i.e. `registry.redhat.io`). Until these credentials have been configured, the builds will continue to fail due to the system being unable to pull the required images.

Reviewer: ```suggestion
Some container builds may use parent images from registries that require authentication, for example, `registry.redhat.io`. Until these credentials have been configured, the builds will continue to fail due to the system being unable to pull the required images.
```
As per the IBM style guide, we should avoid using i.e. , e.g. and any other similar conjunctions.

.Procedure

. Obtain the username and password login credentials for the container registry.
    * For access to `registry.redhat.io`, you can create a registry service account at https://access.redhat.com/terms-based-registry/accounts.

> . In the correct {ProductName} workspace, go to secrets. 
> . Click add secret.  

Reviewer: ```suggestion
. In the correct {ProductName} workspace, go to **Secrets**. 
. Click **Add secret**.  
```
